### PR TITLE
Albrja/mic 5267/build from vbu

### PR DESCRIPTION
**Pull request by albrja**
### Run github actions workflow from vivarium build utils
- *Category*: CI
- *JIRA issue*: https://jira.ihme.washington.edu/browse/MIC-5267

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -9,81 +9,10 @@ on:
...
-          # email subject
-          subject: ${{ github.job }} job of ${{ github.repository }} has ${{ job.status }}
-          # email body as text
-          body: ${{ github.job }} job in worflow ${{ github.workflow }} of ${{ github.repository }} has ${{ job.status }}
-          # comma-separated string, send email to
-          to: uw_ihme_simulationscience@uw.edu
-          # from email name
-          from: Vivarium Notifications
+    uses: 
+      ihmeuw/vivarium_build_utils/.github/workflows/build.yml@albrja/mic-5267/build-from-vbu
```
Comment (stevebachmeier) [inline .github/workflows/build.yml:16]: Everything after the @ would go away when this all gets merged? I assume the default is to use whatever version of the centralized build.yml is on main?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -9,81 +9,10 @@ on:
...
+    with:
+      dependencies: "layered_config_tree"
```
Comment (stevebachmeier) [inline .github/workflows/build.yml:18]: We've lost mypy checking now. Is there a way to inject extra stages in the middle of the other one? Or at the very least just at the end?

**Review by stevebachmeier — CHANGES_REQUESTED** [review]
Same as I said in the build utils comment - I think we need a way to add custom stages, e.g. vivarium needs mypy checking

File: .github/workflows/build.yml (hunk 1 above)
Comment (albrja) [reply .github/workflows/build.yml:16]: Yes. I'd have to merge the VBU PR and then change this to match.

File: CHANGELOG.rst [hunk 3]
```diff
@@ -1,4 +1,4 @@
-**3.0.15 - TBD/TBD/TBD**
+**3.0.15 - 10/24/24**
```
Comment (stevebachmeier) [inline CHANGELOG.rst:1]: Don't forget to include these CI changes!

**Review by stevebachmeier — APPROVED** [review]

File: CHANGELOG.rst (hunk 3 above)
Comment (albrja) [reply CHANGELOG.rst:1]: Yea I still need to remove the branch in the build yml too so once other PR is merged Ill make final updates.

**Review by hussain-jafari — APPROVED** [review]

**Pull request merged**